sync/Dockerfile

```diff
@@ -0,0 +1,8 @@
+FROM busybox:1.36
+
+COPY sync/bin/* .
```

Comment:
Personally, I prefer a standalone docker file. here this depends on the MakeFile 🤔 

Can we first build the binary and then extract that from a staged docker build ?

Reply:
Ya good call. I'll make that change

Reply:
I've done this: https://github.com/open-feature/test-harness/pull/53/commits/841449bf6525cff589a64e8f4c963a28dc9183d9

Reply:
I've removed the go building from the ci.